Automattic/wp-calypso
# Playwright: Migrate the Plans specs and POMs

#### Changes proposed in this Pull Request

Playwright implementation of the old Plans specs in the `specs-calypso` directory.

Specifically, it migrates, with some modification, the coverage from `specs/specs-calypso`...
- `wp-plan-purchase__comparing-spec.js`
- `wp-plan-purchase__renew-spec.js`
- `wp-plan-purchase__upgrade-spec.js`

It does **_not_** include coverage from `wp-plan-purchase__viewing-spec.js`, which is actually a misnomer because it really just covers applying a coupon in the cart. Coupon coverage should be done in a separate, more isolated test case.

These POMs take the new design approach of not using a `BaseContainer` parent class for the POM classes (as outlined in PR #54697 - option A), and instead bakes the needed waiting into class methods.

#### Testing instructions

Run the Playwright tests, and make sure they pass!

Fixes #54842
Implements option A outlined in PR #54697
